README.md

```diff
@@ -71,81 +21,94 @@ All of the files for the project are on the left-hand side. Here's a quick glimp
+Copy and paste the value into your `.env` file.
 
+Fetching credentials is covered in detail in the [linked roles tutorial](https://discord.com/developers/docs/getting-started).
 
-## Try it out
-Now that you've built your bot, let's give it a try both from the server owner and the user's perspective.
+### Running your app
 
-### Creating the linked role
-To try out our bot, we're going to create a linked role in a server where you have admin permissions.  Open up the server settings, select roles, and click on `Create Role`.  Give the role a name, save it, then click on `Links`.  Click the `Add requirement` button, and you should see your bot in the list of available Apps.  Click on it, and you will see a setup screen where you can configure specific criteria for your role.
+After your credentials are added, you can run your app:
 
-![Verification Setup](assets/verification-setup.png "Verification Setup")
+```
+$ node app.js
+```
+
+And, just once, you need to register you connection metadata schema. In a new window, run:
+
+```
+$ node src/register.js
+```
+
+### Set up interactivity
 
-### Acquiring the role
-To acquire your newly created role, click the server name in the upper left corner of the screen, and select `Linked Roles`.  Click on your role, and it will present the opportunity to connect your account.
+The project needs a public endpoint where Discord can send requests. To develop and test locally, you can use something like [`ngrok`](https://ngrok.com/) to tunnel HTTP traffic.
 
-![Connect accounts](assets/connect-account.png "Connect accounts")
+Install ngrok if you haven't already, then start listening on port `3000`:
 
-Click on the linked role criteria. This should lead to the Discord OAuth2 consent screen.  Click `Authorize`, and then return to Discord.  
+```
+$ ngrok http 3000
+```
 
-![Consent Dialog](assets/consent-dialog.png "Consent Dialog")
+You should see your connection open:
 
-After returning to Discord, you should see a confirmed linked role.
+```
+Tunnel Status                 online
+Version                       2.0/2.0
+Web Interface                 http://127.0.0.1:4040
+Forwarding                    http://1234-someurl.ngrok.io -> localhost:3000
+Forwarding                    https://1234-someurl.ngrok.io -> localhost:3000
+
+Connections                  ttl     opn     rt1     rt5     p50     p90
+                              0       0       0.00    0.00    0.00    0.00
+```
 
-![Connected](assets/connected.png "Connected")
+Copy the forwarding address that starts with `https`, in this case `https://1234-someurl.ngrok.io`, then go to your [app's settings](https://discord.com/developers/applications).
 
-Finally, create a new private channel, and add the new linked role.
+On the **General Information** tab, there will be an **Linked Roles Verification URL**. Paste your ngrok address there, and append `/discord-oauth-callback` to it (`https://1234-someurl.ngrok.io/discord-oauth-callback` in the example).
```

Comment:
In the General Information tab, you use `/linked-role`.  In the OAuth2 redirect, you use `/discord-oauth-callback`.